vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/implementation/VLogReasoner.java

```diff
@@ -808,4 +812,42 @@ ReasonerState getReasonerState() {
 	void setReasonerState(ReasonerState reasonerState) {
 		this.reasonerState = reasonerState;
 	}
+
+	@Override
+	public void writeInferredFacts(OutputStream stream) throws IOException {
+		HashSet<Predicate> toBeQueriedHeadPredicates = new HashSet<Predicate>();
+		for (Rule rule : this.knowledgeBase.getRules()) {
+			for (Literal literal : rule.getHead()) {
+				toBeQueriedHeadPredicates.add(literal.getPredicate());
+			}
+		}
+		for (Fact fact : this.knowledgeBase.getFacts()) {
+			if (!toBeQueriedHeadPredicates.contains(fact.getPredicate()))
+				stream.write((fact.toString() + "\n").getBytes());
+		}
+		for (Predicate predicate : toBeQueriedHeadPredicates) {
+			ArrayList<Term> tobeGroundedVariables = new ArrayList<Term>();
+			for (int i = 0; i < predicate.getArity(); i++) {
+				tobeGroundedVariables.add(Expressions.makeUniversalVariable("X" + i));
+			}
+			final QueryResultIterator answers = this
+					.answerQuery(Expressions.makePositiveLiteral(predicate, tobeGroundedVariables), true);
+			answers.forEachRemaining(queryAnswer -> {
+				try {
+					stream.write(Serializer.getFactOutput(predicate, queryAnswer.getTerms()).getBytes());
+				}
+
+				catch (IOException e) {

```

Comment:
never do `e.printStackTrace();` when catching an exception, it is too expensive.
We should propagate this exception (like we do for exportQueryAnswersToCsv), and perhaps log it first.